apps/site/lib/site_web/controllers/schedule/shuttles_controller.ex

```diff
@@ -0,0 +1,23 @@
+defmodule SiteWeb.ScheduleController.ShuttlesController do
+  @moduledoc false
+
+  use SiteWeb, :controller
+
+  alias SiteWeb.ScheduleView
+
+  plug(SiteWeb.Plugs.Route)
+  plug(SiteWeb.ScheduleController.Defaults)
+  plug(:alerts)
+  plug(SiteWeb.Plugs.AlertsByTimeframe)
```

Comment:
The mix of patterns here is a bit hard to follow — these two plugs are related and one depends on the other, but one is implemented as a `Plug` and one as a call to a shared helper function. I wonder if it would be much work to make `assign_alerts` into a `Plug`? It only seems to be used in a few places.

Reply:
I'll look into it.

Reply:
There actually does seem to be a good reason to do this, to wit, we call `assign_alerts` with different parameters from one controller to the other, and I'm not aware of a simple way to implement that with `Plug`s--you'd have to assign the pertinent arguments to a known assign in the `conn` and then pull that back out in the `Plug`. But if I'm missing a better way to do it, I'd like to hear about it.

Reply:
`plug` accepts a second argument, which if present is handed as a second argument to the plug's `init` and `call` functions. For example:

```elixir
plug SiteWeb.Plugs.AssignAlerts, filter_by_direction?: true
```

We are already doing this in `SiteWeb.Plugs.Banner`, but it appears no controller that uses the banner actually uses the options.

Reply:
TIL. OK, seems like a simple refactoring.

Reply:
@digitalcora I see that there is at least one place that uses `assign_alerts` outside of a plug context (i.e. just as a regular function call), and it's not clear to me if changing that to a plug will break the logic, so on reconsideration I'm inclined to leave this as-is.

Reply:
Sounds fine. I'd ordinarily advocate taking a closer look but we should prioritize getting this skeleton in place to be able to build on it.